Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,6 +140,185 @@ public class PushDeploymentController : ApiController
             }
         }
 
+        [HttpPost]
+        [HttpPut]
+        public async Task<HttpResponseMessage> OnePushDeploy(
+            [FromUri] bool isAsync = false,
+            [FromUri] string author = null,
+            [FromUri] string authorEmail = null,
+            [FromUri] string deployer = OneDeploy,
+            [FromUri] string message = DefaultMessage)
+        {
+            using (_tracer.Step("OnePushDeploy"))
+            {
+                var response = Request.CreateResponse();
+
+                //Parses all query parameters into queryCollection
+                var queryCollection = Request.RequestUri.ParseQueryString();
+                var deployType = queryCollection[Constants.Type];
+                var website_framework = System.Environment.GetEnvironmentVariable(Constants.Framework).ToLower();
+                //Checks the "path" queary parameter, path root at /wwwroot TBD if need to change to /home
+                string path = queryCollection[Constants.FilePath];
+
+                //Checks the "clean" query parameter, defauts to False if no query parameter is provided
+                var cleanQuery = queryCollection[Constants.Clean];
+                bool isCleanDeployment = Convert.ToBoolean(cleanQuery);
+
+                //Checks the "async" query parameter, defaults to false if no query parameter is provided
+                var asyncQuery = queryCollection[Constants.Async];
+                isAsync = Convert.ToBoolean(asyncQuery);
+
+                var deploymentInfo = new ZipDeploymentInfo(_environment, _traceFactory)
+                {
+                    AllowDeploymentWhileScmDisabled = true,
+                    Deployer = deployer,
+                    IsContinuous = false,
+                    AllowDeferredDeployment = false,
+                    IsReusable = false,
+                    CleanupTargetDirectory = isCleanDeployment,
+                    TargetChangeset = DeploymentManager.CreateTemporaryChangeSet(message: "Deploying from pushed file"),
```

Comment:
>TargetChangeset [](start = 20, length = 15)

How is this used: TargetChangeset?

Reply:
TargetChangeset is used quite extensively and I can't pin point its exact use but it seems like it tracks changes made to the webapp